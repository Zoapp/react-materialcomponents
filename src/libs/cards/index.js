/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import CardActions from "./actions";
import Card from "./card";
import CardMedia from "./media";
import CardText from "./text";
import CardPrimary from "./primary";
import CardTitle from "./title";
import CardSubtitle from "./subtitle";

export {
  Card,
  CardActions,
  CardMedia,
  CardText,
  CardPrimary,
  CardTitle,
  CardSubtitle,
};
