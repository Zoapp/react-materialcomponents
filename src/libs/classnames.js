/*!
  Copyright (c) 2017 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
  Rewritten to ES7 by Mik BRY
*/
const ClassNames = (args) => {
  const classes = [];
  args.forEach((arg) => {
    if (arg) {
      const argType = typeof arg;
      if (argType === "string" || argType === "number") {
        classes.push(arg);
      } else if (Array.isArray(arg) && arg.length) {
        const inner = ClassNames(arg);
        if (inner) {
          classes.push(inner);
        }
      } else if (argType === "object") {
        Object.keys(arg).forEach((key) => {
          if (arg[key]) {
            classes.push(key);
          }
        });
      }
    }
  });
  return classes.join(" ");
};

export default ClassNames;
