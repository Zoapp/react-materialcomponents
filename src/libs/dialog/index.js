/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Dialog from "./dialog";
import DialogBody from "./body";
import DialogFooter from "./footer";
import DialogHeader from "./header";

export { Dialog, DialogBody, DialogFooter, DialogHeader };
