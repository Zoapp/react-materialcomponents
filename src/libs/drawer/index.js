import Drawer from "./drawer";
import DrawerContent from "./content";
import DrawerHeader from "./header";

export { Drawer, DrawerContent, DrawerHeader };
