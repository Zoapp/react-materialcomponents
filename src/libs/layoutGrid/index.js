/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Grid from "./grid";
import Inner from "./inner";
import Cell from "./cell";

export { Grid, Inner, Cell };
