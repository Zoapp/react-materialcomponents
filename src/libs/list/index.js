/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import ListItem from "./item";
import ListItemMeta from "./meta";
import List from "./list";
import ListDivider from "./divider";
import ListGroup from "./group";
import ListGroupSubheader from "./subheader";

export {
  List,
  ListItem,
  ListItemMeta,
  ListDivider,
  ListGroup,
  ListGroupSubheader,
};
