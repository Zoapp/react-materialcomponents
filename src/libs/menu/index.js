/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import MenuAnchor from "./anchor";
import MenuItem from "./item";
import MenuDivider from "./divider";

import Menu from "./menu";

export { Menu, MenuAnchor, MenuItem, MenuDivider };
