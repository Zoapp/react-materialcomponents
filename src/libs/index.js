/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Rmdc from "./rmdc";
import Button from "./components/button";
import Icon from "./components/icon";
import Content from "./components/content";
import Checkbox from "./components/checkbox";
import Fab from "./components/fab";
import TextField from "./components/textfield";
import Radio from "./components/radio";
import FormField from "./components/formField";
import Select from "./components/select";
import { Menu, MenuAnchor, MenuItem, MenuDivider } from "./menu";
import Snackbar from "./components/snackbar";
import LinearProgress from "./components/linearProgress";
import Switch from "./components/switch";
import Slider from "./components/slider";
import IconToggle from "./components/iconToggle";
import { Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon } from "./toolbar";
import { Drawer, DrawerContent, DrawerHeader } from "./drawer";
import { List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader } from "./list";
import { Grid, Inner, Cell } from "./layoutGrid";
import { Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle } from "./cards";
import { Tabbar, Tab } from "./tab";
import { GridList, Tile } from "./gridList";
import { Dialog, DialogHeader, DialogBody, DialogFooter } from "./dialog";

export {
  Button, Content, Fab, TextField, Checkbox, Radio, FormField, Select, Snackbar,
  LinearProgress, Switch, Slider, IconToggle, Icon,
  Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon,
  Drawer, DrawerContent, DrawerHeader,
  List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader,
  Grid, Inner, Cell,
  Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle,
  Tabbar, Tab,
  GridList, Tile,
  Dialog, DialogHeader, DialogBody, DialogFooter,
  Menu, MenuAnchor, MenuItem, MenuDivider,
};

export default Rmdc;
