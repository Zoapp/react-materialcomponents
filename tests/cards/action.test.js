import React from "react";
import renderer from "react-test-renderer";
import { CardActions, Button } from "libs/";

describe("cards/CardActions", () => {
  it("can render correctly", () => {
    const tree = renderer
      .create(
        <CardActions>
          <Button>Click me!</Button>
        </CardActions>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can be verticaly aligned", () => {
    const tree = renderer.create(<CardActions vertical />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
