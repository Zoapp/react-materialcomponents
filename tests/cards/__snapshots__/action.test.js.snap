// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/cards/CardActions can be verticaly aligned 1`] = `
<section
  className="mdc-card__actions mdc-card__actions--vertical"
/>
`;

exports[`components/cards/CardActions can render correctly 1`] = `
<section
  className="mdc-card__actions"
>
  <button
    className="mdc-button mdc-button--compact mdc-card__action"
    disabled={false}
    id={null}
    onClick={null}
    type="button"
  >
    
    Click me!
  </button>
</section>
`;
